ENIGMAsystem/SHELL/Universal_System/Extensions/FileManip/Universal/fileman.cpp

```diff
@@ -0,0 +1,402 @@
+﻿/*
+
+ MIT License
+ 
+ Copyright © 2020 Samuel Venable
+ 
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+ 
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+ 
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+ 
+*/
+
+#include "../fileman.h"
+#include <filesystem>
+#include <iostream>
+#include <sstream>
+#include <set>
+
+namespace fs = std::filesystem;
+
+using namespace strings;
+
+using std::string;
+using std::vector;
+using std::size_t;
+using std::cout;
+using std::endl;
+
+namespace strings {
+
+  string string_replace_all(string str, string substr, string nstr) {
+    size_t pos = 0;
+    while ((pos = str.find(substr, pos)) != string::npos) {
+      str.replace(pos, substr.length(), nstr);
+      pos += nstr.length();
+    }
+    return str;
+  }
+
+  vector<string> split_string(const string &str, char delimiter) {
+    vector<string> vec;
+    std::stringstream sstr(str);
+    string tmp;
+    while (std::getline(sstr, tmp, delimiter))
+      vec.push_back(tmp);
+    return vec;
+  }
+
+  string filename_path(string fname) {
+    size_t fp = fname.find_last_of("/\\");
+    return fname.substr(0,fp + 1);
+  }
+
+  string filename_name(string fname) {
+    size_t fp = fname.find_last_of("/\\");
+    return fname.substr(fp + 1);
+  }
+
+  string filename_ext(string fname) {
+    fname = filename_name(fname);
+    size_t fp = fname.find_last_of(".");
+    if (fp == string::npos)
+      return "";
+    return fname.substr(fp);
+  }
+
+} // namespace strings
+
+namespace fileman {
+
+  string get_working_directory_ns() {
+    string result = "";
+    try {

```

Comment:
This is likely to miss other exceptions; you should favor the noexcept versions.